Implement an aiokatcp client and server.

ijassiem opened the pull request:
A server class is created which derives from the aiokatcp DeviceServer class. The client is implemented using an object from the aiokatcp.Client class. The server responds accordingly to requests made by the client.

This code demonstrates the use of the aiokatcp module for creating an aiokatcp client and server, and
forms part of the DSP training.

Contributes to: DGDT-48

ijassiem committed:
Implement an aiokatcp client and server.

A server class is created which derives from the aiokatcp DeviceServer class. The client is implemented using an object from the aiokatcp.Client class. The server responds accordingly to requests made by the client.

This code demonstrates the use of the aiokatcp module for creating an aiokatcp client and server, and
forms part of the DSP training.

Contributes to: DGDT-48

james-smith-za commented on aiokatcp/aiokatcp_client.py:
```diff
@@ -0,0 +1,26 @@
+PORT = 4444
+
+async def run_client():
+    """Co-routine starts an aiokatcp client with requests to server."""
```
My only suggestion is to change this to imperative mood, and be a bit more descriptive as to what the function actually does. Something like this perhaps:

```suggestion
    """Connect to a katcp server, and make some requests.
    
    Request the time, print the output of `?help` and show the echo functionality.
    """
```

james-smith-za commented on aiokatcp/aiokatcp_server.py:
```diff
@@ -0,0 +1,44 @@
+import datetime
+
+class MyServer(aiokatcp.DeviceServer):
+    """This class inherits from the aiokatcp DeviceServer class.
+
+    This class is used to create an aiokatcp server object.
+    """
```
Be a bit more descriptive in your documentation what your derived class actually adds, or why it's created. Saying that it's inheriting from DeviceServer adds little value to your documentation because I can see in the line just above that this is what is happening. I'd like to know what you do in this derived class that the base doesn't do.

In this case, it's part of a training exercise so there's no purpose, but it may be something along the lines of "Implement a control and monitoring interface for a correlator."

My suggestion of how to implement this in your specific case would look something like this perhaps:
```suggestion
    """Example katcp server derivation for training purposes.

     Some example requests are added to the base class to demonstrate the format for the
     benefit of the trainee.
    """
```

james-smith-za reviewed (commented):
Looking very good! Code is clear, concise, just the docstrings that I can see which could use a bit of improvement.

ijassiem committed:
Add more descriptive docstrings.

These changes relates to improvements suggested in the pull request: Implement an aiokatcp client and server. #7

Contributes to: DGDT-48
